File: Randnm/randnm.py
import numpy
import random
import names
import surnames
import set
   
def randpos(arr):
    return arr[int(numpy.floor(random.random()*(len(arr)-1)))]
   
def randcode(mlen, rlen):
    leng = mlen + numpy.floor(random.random()*rlen)
    res = ""
    while leng > 0:
        leng = leng-1
        res = res + randpos(set.data)
    print("code: ", res)
def randate(minage, curyear):
    year = minage+randpos(range(0,minage))
    month = month = 1+randpos(range(0,12))
    day = 0
    if month == 2:
        day = day = 1+randpos(range(0,28))
    elif month % 2 == 1:
        day = 1+randpos(range(0,31))
    else:
        day = 1+randpos(range(0,30))
    print("date: ", day, ".", month, ".", curyear-year)
def randnm():
    name = randpos(names.names)
    surname = randpos(surnames.surnames)
    print("Name: ", name, surname)
randcode(8, 4)
randate(20, 2023)
randnm()
 

File: Randnm/web/Randnm.js
let set = `qwertyuiopasdfghjklzxcvbnm1234567890`
   function randlen() {
       return 8+Math.floor(Math.random()*4)
       }
   function randcode(len=randlen()) {
       var res = ""
       for(var i = 0; i < len; i++) {
           res += set[Math.floor(Math.random()*(set.length-1))]
           }
       return res
       };
   function randate(minage=20, curryear=2023) {
     var year = minage + Math.floor(Math.random()*minage),month = 1+ Math.floor(Math.random()*11), day
     if(month == 2) { day = 1 + Math.floor(Math.random()*27) }
     else if(month % 2 == 1) { day = 1+Math.floor(Math.random()*30) }
     else { day = 1+Math.floor(Math.random()*29) }
     return { day:  day, month: month, year: curryear-year}
     }
   function randnm() {
       var name = names[Math.floor(Math.random()*(names.length-1))]
       var surname = surnames[Math.floor(Math.random()*(surnames.length-1))]
       return { name: name, surname: surname }
      }
 
